docs/quickstart.rst

```diff
@@ -129,6 +129,32 @@ certain number of old archives::
     borg prune -v $REPOSITORY --prefix '{hostname}-' \
         --keep-daily=7 --keep-weekly=4 --keep-monthly=6
 
+Pitfalls with shell variables and environment variables
+-------------------------------------------------------
+
+This applies to all environment variables you want borg to see, not just
+``BORG_PASSPHRASE``. The short explanation is: always ``export`` your variable,
+and use single quotes if you're unsure of the details of your shell's expansion
+behavior. E.g.::
+
+    export BORG_PASSPHRASE='complicated & long';
+
+This ensures the borg process that needs to access this passphrase can do so.
+Note that this is not the same thing as::
+
+  $ BORG_PASSPHRASE='complicated & long'; # missing "export"!
+  $ ./yourborgjob.sh # will never see the above $BORG_PASSPHRASE value
+
+This is because ``export`` exposes variables to subprocesses, which borg may be
+one of. More on ``export`` can be found in the "ENVIRONMENT" section of the
+bash(1) man page.
+
+.. note:: ``sudo`` does not inherit your environment, so ``sudo borg`` will not
+          have access to ``BORG_PASSPHRASE`` despite running the above example.
+
+To debug what your borg process is actually seeing, find its PID
+(``ps aux|grep borg``) and then look into ``/proc/<PID>/environ``.
```

Comment:
it might be more confusing, but i don't think it hurts to just submit, and wait to hear feedback from folks getting confused or not?

Wrt, "level of help" - I knew about `/proc/` yet still didn't think to do this on my own (someone in IRC suggested it). That is: the helpfulness of this hint in the documentation is not necessarily related to being programmer.